### PR TITLE
Move helper functions from cpu/common/libbkrm.sh to cki_lib/libcki.sh

### PR DESCRIPTION
**Pull request by idorax**
Now test cases in both `power-management/*` and `cpu/*` share a same library file `cki_lib/libcki.sh`.

### CODE REVIEW

**Comment by idorax** [comment]
[PR#342](https://github.com/CKI-project/tests-beaker/pull/342) is closed because I renamed the branch from `dev.0830.cki_lib.update` to `dev.0830.CKILIBUPDATE`.

File: cki_lib/libcki.sh [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+#
+function cki_set_reason()
+{
+    typeset rc=${1?"*** result code"}
+    shift
+    case $rc in
+    $CKI_FAIL) g_reason_fail="$@" ;;
+    $CKI_UNSUPPORTED) g_reason_unsupported="$@" ;;
+    $CKI_UNINITIATED) g_reason_uninitiated="$@" ;;
+    *) ;;
```
Comment (emanchado) [inline cki_lib/libcki.sh]: Maybe not ignore unexpected values in here, either?

File: cpu/common/libutil.sh [hunk 2]
```diff
@@ -17,19 +17,20 @@
...
 function is_rhel7
 {
-    [ ! -e /etc/redhat-release ] && return 0
-    rel=$(cat /etc/redhat-release)
-    ver=$(echo "${rel##*release}" | awk '{$1=$1;print}' | cut -d '.' -f 1)
-    [ "$ver" -eq 7 ] && return 1 || return 0
+    [[ -e /etc/redhat-release ]] && return 0
+
+    typeset rel=$(cat /etc/redhat-release)
+    typeset ver=$(echo "${rel##*release}" | awk '{$1=$1;print}' | \
```
Comment (emanchado) [inline cpu/common/libutil.sh]: I know this was there from before, but: what does `$1=$1` do in awk?

File: cpu/die/runtest.sh [hunk 3]
```diff
@@ -17,35 +17,32 @@
...
 NAME=$(basename $FILE)
 CDIR=$(dirname $FILE)
 TEST=${TEST:-"$0"}
 TMPDIR=/var/tmp/$(date +"%Y%m%d%H%M%S")
 
-source ${CDIR%/cpu/die}/cpu/common/libbkrm.sh
-source ${CDIR%/cpu/die}/cpu/common/libutil.sh
-source ${CDIR%}/../../cki_lib/lib.sh
+#source /usr/bin/rhts_environment.sh
+#source /usr/share/beakerlib/beakerlib.sh
```
Comment (emanchado) [inline cpu/die/runtest.sh]: Could we remove these two commented-out lines?

File: cpu/driver/runtest.sh [hunk 4]
```diff
@@ -23,74 +23,75 @@ CDIR=$(dirname $FILE)
 TEST=${TEST:-"$0"}
 TMPDIR=/var/tmp/$(date +"%Y%m%d%H%M%S")
 
-source ${CDIR%/cpu/driver}/cpu/common/libbkrm.sh
-source ${CDIR%/cpu/driver}/cpu/common/libutil.sh
+#source /usr/bin/rhts_environment.sh
+#source /usr/share/beakerlib/beakerlib.sh
```
Comment (emanchado) [inline cpu/driver/runtest.sh]: Same here.

File: cpu/idle/runtest.sh [hunk 5]
```diff
@@ -23,45 +23,46 @@ CDIR=$(dirname $FILE)
 TEST=${TEST:-"$0"}
 TMPDIR=/var/tmp/$(date +"%Y%m%d%H%M%S")
 
-source ${CDIR%/cpu/idle}/cpu/common/libbkrm.sh
-source ${CDIR%/cpu/idle}/cpu/common/libutil.sh
+#source /usr/bin/rhts_environment.sh
+#source /usr/share/beakerlib/beakerlib.sh
```
Comment (emanchado) [inline cpu/idle/runtest.sh]: Same here: remove if unneeded?

File: cki_lib/libcki.sh [hunk 6]
```diff
@@ -0,0 +1,229 @@
...
+    else
+        report_result ${TEST} FAIL 1
+    fi
+    rhts-abort -t recipe
+    exit $CKI_STATUS_ABORTED
+}
+
+function cki_abort_task()
+{
+    rlLog "Aborting current task because $*"
```
[the remark this answers is not in the document]
Comment (idorax) [reply cki_lib/libcki.sh]: Replaced.

File: cki_lib/libcki.sh [hunk 7]
```diff
@@ -0,0 +1,229 @@
...
+    rhts-report-result "$TEST" WARN "$OUTPUTFILE"
+    exit $CKI_STATUS_ABORTED
+}
+
+function cki_skip_task()
+{
+    rlLog "Skipping current task because $*"
```
[the remark this answers is not in the document]
Comment (idorax) [reply cki_lib/libcki.sh]: Done.

File: cki_lib/libcki.sh [hunk 8]
```diff
@@ -0,0 +1,229 @@
...
+{
+    typeset rc=${1?"*** result code"}
+    shift
+    typeset argv="$*"
+    case $rc in
+    $CKI_PASS) rlPass $argv ;;
+    $CKI_FAIL) rlFail "$g_reason_fail #$argv#" ;;
+    $CKI_UNSUPPORTED) cki_skip_task  "$g_reason_unsupported #$argv#" ;;
+    $CKI_UNINITIATED) cki_abort_task "$g_reason_uninitiated #$argv#" ;;
+    *) ;;
```
[the remark this answers is not in the document]
Comment (idorax) [reply cki_lib/libcki.sh]: Added `cki_abort_task ...`.

File: cki_lib/libcki.sh (hunk 1 above)
Comment (idorax) [reply cki_lib/libcki.sh]: Added `*) g_reason_other="$rc is an invalid result code" ;;`.

File: cpu/die/runtest.sh (hunk 3 above)
Comment (idorax) [reply cpu/die/runtest.sh]: Removed.

File: cpu/driver/runtest.sh (hunk 4 above)
Comment (idorax) [reply cpu/driver/runtest.sh]: Removed.

File: cpu/idle/runtest.sh (hunk 5 above)
Comment (idorax) [reply cpu/idle/runtest.sh]: Removed.

File: cpu/common/libutil.sh (hunk 2 above)
Comment (idorax) [reply cpu/common/libutil.sh]: Well, it looks weird. I updated it via sed + awk.
```patch
-    typeset rel=$(cat /etc/redhat-release)
-    typeset ver=$(echo "${rel##*release}" | awk '{$1=$1;print}' | \
-                  cut -d '.' -f 1)
+    typeset ver=$(sed 's/.*release//' /etc/redhat-release | \
+                  awk '{print $1}' | \
+                  awk -F'.' '{print $1}')
```
BTW, the patch is tested on Fedora, RHEL8 and RHEL7 manually. For test log, please refer to:
* http://pastebin.test.redhat.com/798166

**Comment by idorax** [comment]
Test log:
1. **Boot test** : `J:3790050`
1. **cpu/die**: http://10.66.13.186/t/sandbox/ark/PR369/testlog01.cpu.die.txt
1. **cpu/driver**: http://10.66.13.186/t/sandbox/ark/PR369/testlog02.cpu.driver.txt
1. **cpu/idle**: http://10.66.13.186/t/sandbox/ark/PR369/testlog03.cpu.idle.txt
1. **power-management/cpupower/sanity**: http://10.66.13.186/t/sandbox/ark/PR369/testlog04.power-management.cpupower.txt
1. **power-management/rapl/powercap**: http://10.66.13.186/t/sandbox/ark/PR369/testlog05.power-management.rapl.txt

**NOTE**: 

- All test cases are manually run because it's not easy to reserve a machine having intel CPU via beaker, so I manually reserved a system and run cases on it.
- `power-management/rapl/powercap` FAIL, but it is not caused by changes of this PR.

**Comment by idorax** [comment]
Test cases are executed via beaker on the system I once reserved now, the job id is `J:3793515`.

**Comment by idorax** [comment]
Thanks @emanchado for review again. Looks I don't have permissions to merge this PR. If you have permissions, please help to merge it, else I'll ask @veruu or @rasibley  for help.

**Pull request merged**